### PR TITLE
Added support to upload mapping.txt files

### LINKED ISSUE #14: Add support for uploading ProGuard map files
Add ability to upload ProGuard obfuscation map files, what google in the developer console calls "deobfuscation files".

### PR DESCRIPTION
**Pull request by lkillgore**
Fixes #14.  Adapted from the PR #128.

image

### CODE REVIEW

**Comment by lkillgore** [comment]
Hi @NxSoftware, could you please take a look?  Thanks!

File: Tasks/google-play-release/GooglePlay.ts [hunk 1]
```diff
@@ -108,13 +108,29 @@ async function run() {
...
                 const apk: googleutil.Apk = await googleutil.addApk(edits, packageName, apkFile);
                 tl.debug(`Uploaded ${apkFile} with the version code ${apk.versionCode}`);
                 apkVersionCodes.push(apk.versionCode);
+
+                if (tryUploadMappingFile) {
+                    const mappingFilePattern = tl.getPathInput('mappingFilePath', false);
+                    tl.debug(`Mapping file pattern: ${mappingFilePattern}`);
+
+                    const mappingFilePath = resolveGlobPath(mappingFilePattern);
+                    tl.checkPath(mappingFilePath, 'mappingFilePath');
```
Comment (NxSoftware) [inline Tasks/google-play-release/GooglePlay.ts]: I'm not familiar with this API. What happens if the glob pattern finds multiple mapping.txt files? e.g. `**/mapping.txt` finds `apk1/mapping.txt` and `apk2/mapping.txt`. Does it always return the first one?

My original implementation made an assumption that the path directly specifies the mapping.txt file related to the main APK, and I believe that's what will happen here too but wanted to check.

**Comment by NxSoftware** [comment]
> Hi @NxSoftware, could you please take a look? I tested this and uploaded a `mapping.txt` file, which seems to upload w/o any errors. How did you go about testing your change? Other than uploading it, I'm not sure how to verify that google play will correctly 'deobfuscate'. Thoughts?
> 
> Thanks!

I had an Android developer friend run a build on a pipeline where this task was installed as a preview via `tfx-cli`. He has a special debug button in an Android app that causes a crash which showed up in Google Play with the de-obfuscated stack trace.

File: Tasks/google-play-release/GooglePlay.ts (hunk 1 above)
Comment (lkillgore) [reply Tasks/google-play-release/GooglePlay.ts]: It returns the one.  It uses the `glob.sync()` method and returns the first result.  You can also specify the full path to the mapping as in your original implementation.  I figure that 'most' projects will just have the one APK file, and likely also just have the one mapping file, so allowing glob results would help in set up for the common case.

Comment (NxSoftware) [reply Tasks/google-play-release/GooglePlay.ts]: Sounds good to me!

File: Tasks/google-play-release/GooglePlay.ts [hunk 2]
```diff
@@ -108,13 +108,29 @@ async function run() {
...
+                    console.log(tl.loc('FoundDeobfuscationFile', mappingFilePath));
+                    tl.debug(`Uploading mapping file ${mappingFilePath}`);
+                    await googleutil.uploadDeobfuscation(edits, mappingFilePath, packageName, apk.versionCode);
+                    tl.debug(`Uploaded ${mappingFilePath} for APK ${apkFile}`);
+
+                    // Only upload the first / main APK
+                    tryUploadMappingFile = false;
```
Comment (yacaovsnc) [inline Tasks/google-play-release/GooglePlay.ts]: It seems the `packageName` is set from outside the loop, so we just need `apk.versionCode` here.  Another option is to just set `apk.versionCode` within this look, and move the upload mapping file block outside.  This way we don't have to set a boolean to control the flow, and it's clear that we will only upload the mapping file once.

Comment (lkillgore) [reply Tasks/google-play-release/GooglePlay.ts]: I like it!

**Pull request merged**